doc/api/stream.md

```diff
@@ -443,9 +443,33 @@ first argument. To reliably detect write errors, add a listener for the
 The return value is `true` if the internal buffer does not exceed
 `highWaterMark` configured when the stream was created after admitting `chunk`.
 If `false` is returned, further attempts to write data to the stream should
-stop until the [`'drain'`][] event is emitted. However, the `false` return
-value is only advisory and the writable stream will unconditionally accept and
-buffer `chunk` even if it has not not been allowed to drain.
+stop until the [`'drain'`][] event is emitted.
+
+In case the `false` return value is ignored, the writable stream will
+unconditionally accept and buffer `chunk` even if it has not not been allowed
+to drain. However, Node has a limited amount of memory, and not
+listening to `'drain'` might cause the process to abort by an out of
+memory error, and other types of memory and garbage collection
+issues, such as high RSS.
+
+The `'drain'` event must be handled if the writable is a [Transform][],
+because the `Transform` streams are paused by default until
+they are piped or an `'data'` or `'readable'` event handler is added.
+So, ignoring when `write()` returns false will cause the data to be buffered
+until the stream is flowing, causing a memory leak.
```

Comment:
I suppose `Transform` streams *cannot* flow until they are connected to consumers, but the same issues exist even for writable FS/network streams because you never know when they’ll start draining. Especially if the other end is a TCP server, the TCP server could just never read anything from its end and cause the exact same problem once the OS-level sockets buffers fill up. So I’m not convinced this is actually a `Transform`-specific thing, but I think I’m alone with that opinion. Also, it’s not a memory leak unless you know for certain that the `Transform` stream will *never* be piped to something else in the future. E.g., you could be buffering an entire file in-memory on purpose by using a `Transform` stream with the identity transform ;-).

I don’t understand the RSS issue, though, does expanding a `Buffer` and then letting it get GCed at a later point really cause `node` to allocate memory without ever freeing it? Again, my concerns are probably dismissable.